docs/proposals/20200220-cluster-resource-set.md

```diff
@@ -0,0 +1,208 @@
+     * [Risks and Mitigations](#risks-and-mitigations)
+  * [Alternatives](#alternatives)
+  * [Upgrade Strategy](#upgrade-strategy)
+  * [Additional Details](#additional-details)
+     * [Test Plan [optional]](#test-plan-optional)
+     * [Graduation Criteria [optional]](#graduation-criteria-optional)
+  * [Implementation History](#implementation-history)
+
+  
+## Glossary
+
+Refer to the [Cluster API Book Glossary](https://cluster-api.sigs.k8s.io/reference/glossary.html).
+
+## Summary
+
+Provide a mechanism for applying resources in a cluster once it is created.
+
+## Motivation
+
+Clusters created by Cluster API are minimally functional. For instance,they do not have a container networking interface (CNI), which is required for pod-to-pod networking, or any StorageClasses, which are required for dynamic persistent volume provisioning.
+Users today must manually add these components to every cluster they create.
+
+Having a mechanism to apply an initial set of default resources after clusters are created makes clusters created with Cluster API functional and ready for workloads from the beginning, without requiring additional user intervention.
+
+To achieve this, ClusterResourceSet CRD is introduced that will be responsible for applying a set resources defined by users to the matching clusters.
+
+### Goals
+
+- Provide a means to specify a set of resources to apply automatically to newly-created and existing Clusters. Resources will be applied only once.
+- Support additions to the resource list by applying the new added resources to both new and existing matching clusters.
+- Support both json and yaml resources.
+
+### Non-Goals/Future Work
+
+- Replace or compete with the Cluster Addons subproject.
+- Support deletion of resources from clusters. Deleting a resource from a ClusterResourceSet or deleting a ClusterResourceSet does not result in deletion of those resources from clusters.
+- Lifecycle management of the installed resources (such as CNI).
+- Support reconciliation of resources on resource hash change and/or periodically. This can be a future enhancement work.
+
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As someone creating multiple clusters, I want all my clusters to have a CNI provider of my choosing installed automatically, so I don’t have to manually repeat the installation for each new cluster.
+
+#### Story 2
+
+As someone creating multiple clusters, I want all my clusters to have a StorageClass installed automatically, so I don't have to manually repeat the installation for each new cluster.
+
+#### Story 3
+
+As someone creating multiple clusters, I want to be able to provide different values for some fields in the resources for different clusters. For example, CNIs podCIDRs  may be required to be distinct for each cluster, hence some templating mechanism for variable substitution in the resources is needed.
+
+### Implementation Details/Notes/Constraints
+
+#### Data model changes
```

Comment:
nit: 
```suggestion
#### Data model changes to existing API types
```

Since the addition of a new (experimental) type is in itself a data model change